include/mega/crypto/sodium.h

```diff
@@ -78,22 +73,90 @@ class MEGA_API EdDSA
 
     /**
      * @brief Generates a new Ed25519 private key seed. The key seed is stored
-     * in the object.
+     * in the object and, optionally, in the pointer passed as parameter (if not NULL)
      *
      * @param privk Private key seed to return, unless NULL.
      * @return 1 on success, 0 on failure.
      */
-    int genKeySeed(unsigned char* privKey = NULL);
+    int genKeySeed();
+
+    int genKeys();
+
+private:
+    static const int PRIVATE_KEY_LENGTH = crypto_sign_SECRETKEYBYTES;
+    unsigned char* privKey; // don't use it externally, use keySeed instead
+};
+
+
+/**
+ * @brief Asymmetric cryptographic for chat messages encryptiong using
+ * ECDH approach with x25519 key pair.
+ */
+class MEGA_API ECDH
+{
+public:
+    static const int PRIVATE_KEY_LENGTH = crypto_box_SECRETKEYBYTES;
+    static const int PUBLIC_KEY_LENGTH = crypto_box_PUBLICKEYBYTES;
+
+    // TLV key to access to the corresponding value in the TLV records
+    static const string TLV_KEY;
+
+    unsigned char* privKey;
+    unsigned char* pubKey;
+
+    ECDH(unsigned char * privKey = NULL);
+    ~ECDH();
+
+    /**
+     * @brief genKeys Generate a new key pair
+     *
+     * @return 1 on success, 0 on failure.
+     */
+    int genKeys();
 
     /**
-     * @brief Derives the Ed25519 public key from the stored private key seed.
+     * @brief encrypt Encrypt a message using the public key of recipient, the
+     * private key of the sender and a nounce (number used once)
+     *
+     * @param encmsg Encrypted text after encryption. This function ensures that the
+     * first crypto_box_ZEROBYTES bytes of the encrypted text msg are all 0.
+     * @param msg Message to be encrypted. Caller must ensure that the first
+     * crypto_box_ZEROBYTES bytes of the message msg are all 0.
+     * @param msglen Lenght of the message to be encrypted.
+     * @param nounce Number used once. The same nonce must never be used to encrypt another
+     * packet from the sender's private key to this receiver's public key or viceversa.
+     * @param pubKey Public key of the receiver.
+     * @param privKey Private key of the sender.
      *
-     * @param pubKey Public key.
      * @return 1 on success, 0 on failure.
      */
-    int publicKey(unsigned char* pubKey);
-};
+    int encrypt(unsigned char* encmsg, const unsigned char* msg,
+               const unsigned long long msglen, const unsigned char* nounce,
```

Comment:
It's a `nonce`, not a `nounce`:
https://en.wikipedia.org/wiki/Cryptographic_nonce